src/coreclr/jit/instr.cpp

```diff
@@ -1861,9 +1861,12 @@ instruction CodeGenInterface::ins_Load(var_types srcType, bool aligned /*=false*
             else
                 ins = INS_lh;
         }
-        else if (TYP_INT == srcType)
+        else if (varTypeIsInt(srcType))
         {
-            ins = INS_lw;
+            if (varTypeIsUnsigned(srcType))
+                ins = INS_lwu;
+            else
+                ins = INS_lw;
```

Comment:
Small ints on RISC-V are type-extended to 32 bits, then sign-extended to full register length. So sign-extending uint was correct.